app/plugin_api.go

```diff
@@ -289,7 +289,16 @@ func (api *PluginAPI) CreateSession(session *model.Session) (*model.Session, *mo
 }
 
 func (api *PluginAPI) ExtendSessionExpiry(sessionID string, expiresAt int64) *model.AppError {
-	return api.app.extendSessionExpiry(sessionID, expiresAt)
+	session, err := api.app.ch.srv.userService.GetSessionByID(sessionID)
+	if err != nil {
+		return model.NewAppError("extendSessionExpiry", "app.session.get_sessions.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	if err := api.app.ch.srv.userService.ExtendSessionExpiry(session, expiresAt); err != nil {
+		return model.NewAppError("extendSessionExpiry", "app.session.extend_session_expiry.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
```

Comment:
Nit: might be better to keep the `*App.extendSessionExpiry` method and move these 2 calls inside that. So that from the plugin api side, things remain unchanged, but the internal app logic changes.

Reply:
I like of like the plugin api method as that ensure `ExtendSessionExpiry` is the only method named that way and there is no confusion on whether the app or the userService method should be called.